sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsError.cs

```diff
@@ -30,5 +68,55 @@ internal TextAnalyticsError(string code, string message, string target = null)
         /// the property in error).
         /// </summary>
         public string Target { get; }
+
+        /// <summary> Generated Error code. </summary>
+        internal ErrorCodeValue Code { get; }
+        /// <summary> Generated inner error contains more specific information. </summary>
+        internal InnerError Innererror { get; }
+        /// <summary> Generated details about specific errors that led to this reported error. </summary>
+        internal IReadOnlyList<TextAnalyticsError> Details { get; }
+
+        internal static TextAnalyticsError DeserializeTextAnalyticsError(JsonElement element)
+        {
+            string errorCode = default;
+            string message = default;
+            string target = default;
+            TextAnalyticsError innerError = default;
+
+            foreach (JsonProperty property in element.EnumerateObject())
+            {
+                if (property.NameEquals("code"))
+                {
+                    errorCode = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("message"))
+                {
+                    message = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("target"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    target = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("innererror"))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        continue;
+                    }
+                    innerError = DeserializeTextAnalyticsError(property.Value);
+                    continue;
+                }
+            }
+
+            // Return the innermost error, which should be only one level down.
+            return innerError.ErrorCode == default ? new TextAnalyticsError(errorCode, message, target) : innerError;
```

Comment:
What if the innerError has another innerError?

Reply:
The service only supports one innerError. In the case they start supporting more, we return the innermost error